Assets/LeapMotion/Core/Scripts/Attributes/ImplementsInterface.cs

```diff
@@ -45,28 +73,48 @@ public class ImplementsInterfaceAttribute : CombinablePropertyAttribute,
                                                          property.objectReferenceValue);
 
         if (implementingObject == null) {
-          Debug.LogError(property.objectReferenceValue.GetType().Name + " does not implement " + type.Name);
+          Debug.LogError(property.objectReferenceValue.GetType().Name + " does not implement " + _type.Name);
         }
 
         property.objectReferenceValue = implementingObject;
       }
     }
 
+    /// <summary>
+    /// Returns the interface type that this attribute uses to constrain
+    /// object fields that it decorates.
+    /// </summary>
+    public Type GetInterfaceType() {
+      if (this._type != null) {
+        return _type;
+      }
+      else { // use _typeGetterMethodName.
+        var target = targets.Query().FirstOrDefault();
+        var objectType = target.GetType();
+
+        var method = objectType.FindMethodByName(_typeGetterMethodName);
```

Comment:
Might be nice to have some more exact method signature requirements enforced here.  Like the method must be static, must take zero parameters, and must not be generic, and must return a value of type Type.  Right now if the method returns something other than type, there would be an unclear null reference exception somewhere.

Reply:
Don't see a reason for requiring the method to be static. I should enforce returning Type, but it should be fine if the method is generic as well?
Also wouldn't the only requirement be that it doesn't have non-optional arguments?

Reply:
I'm not sure how you would go about invoking a generic method, how would you determine the generic parameter types to use?  And yeah I forgot we have a reference to target, so totally fine to have instance method instead of static.